app/Http/Controllers/Three/PostsController.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Rogue\Http\Controllers\Three;
+
+use Rogue\Models\Post;
+use Illuminate\Http\Request;
+use Rogue\Services\PostService;
+use Rogue\Repositories\SignupRepository;
+use Rogue\Http\Transformers\PostTransformer;
+use Rogue\Http\Controllers\Api\ApiController;
+use Rogue\Http\Controllers\Traits\PostRequests;
+
+class PostsController extends ApiController
+{
+    use PostRequests;
+
+    /**
+     * The post service instance.
+     *
+     * @var \Rogue\Services\PostService
+     */
+    protected $posts;
+
+    /**
+     * The signup repository instance.
+     *
+     * @var \Rogue\Repositories\SignupRepository
+     */
+    protected $signups;
+
+    /**
+     * @var \League\Fractal\TransformerAbstract;
+     */
+    protected $transformer;
+
+    /**
+     * Create a controller instance.
+     *
+     * @param  PostContract  $posts
+     * @return void
+     */
+    public function __construct(PostService $posts, SignupRepository $signups)
+    {
+        $this->posts = $posts;
+        $this->signups = $signups;
+
+        $this->transformer = new PostTransformer;
```

Comment:
Can we create the transformer in the same way as the `PostService` and `SignupRepository` or is it different?

Reply:
@katiecrane great q! Now that I'm looking back on this code, I'm having trouble seeing how the transformer gets created in `PostService` and `SignupRepository` :-/ Do you know where this happens? If I comment out this line (47), it breaks though.

Reply:
@chloealee ah I meant can we create it in the same way that we create the `PostService` in this particular block, not in the `PostService.php` file itself

Reply:
@katiecrane ohohoh I see now! Yes it does! Changed :)